src/main/java/io/quarkus/bot/PingWhenNeedsTriageRemoved.java

```diff
@@ -0,0 +1,125 @@
+package io.quarkus.bot;
+
+import io.quarkiverse.githubapp.ConfigFile;
+import io.quarkiverse.githubapp.event.Issue;
+import io.quarkus.bot.config.QuarkusBotConfig;
+import io.quarkus.bot.config.QuarkusBotConfigFile;
+import io.quarkus.bot.el.SimpleELContext;
+import io.quarkus.bot.util.Labels;
+import io.quarkus.bot.util.Patterns;
+import io.quarkus.bot.util.Strings;
+import org.jboss.logging.Logger;
+import org.kohsuke.github.GHEventPayload;
+import org.kohsuke.github.GHIssue;
+import org.kohsuke.github.GHLabel;
+
+import javax.el.ELContext;
+import javax.el.ELManager;
+import javax.el.ExpressionFactory;
+import javax.el.ValueExpression;
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.Set;
+import java.util.TreeSet;
+
+public class PingWhenNeedsTriageRemoved {
+    private static final Logger LOG = Logger.getLogger(PingWhenNeedsTriageRemoved.class);
+
+    @Inject
+    QuarkusBotConfig quarkusBotConfig;
+
+    void pingWhenNeedsTriageRemoved(@Issue.Unlabeled GHEventPayload.Issue issuePayload,
+            @ConfigFile("quarkus-bot.yml") QuarkusBotConfigFile quarkusBotConfigFile) throws IOException {
+
+        if (quarkusBotConfigFile == null) {
+            LOG.error("Unable to find triage configuration.");
+            return;
+        }
+
+        GHIssue issue = issuePayload.getIssue();
+        GHLabel removedLabel = issuePayload.getLabel();
+        Set<String> mentions = new TreeSet<>();
+
+        for (QuarkusBotConfigFile.TriageRule rule : quarkusBotConfigFile.triage.rules) {
+            if (matchRule(issue, rule)) {
```

Comment:
The logic should be different here. The bot already tried to apply the rules on title and body and wasn't able to find anything useful (thus the `needs-triage` label) so doing it again won't do anything more.

See what I described here: https://github.com/quarkusio/quarkus-bot/issues/26#issue-820200347 . Here what we want is find people to mention based on labels that were manually added by someone prior to removing the `needs-triage` label. So you should find the persons to ping based on the labels.

Reply:
Yes, got the point. `needs-triage` label was already there, it means rules have already been applied but no useful information is retrieved.

@gsmet  So probably I should modify my method `matchRule` to apply the rule only on the labels and based on that ping the right person? :)